perfkitbenchmarker/linux_packages/maven.py

```diff
@@ -0,0 +1,79 @@
+import posixpath
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker.linux_packages import INSTALL_DIR
+
+try:
+  import urlparse
+except ImportError:
+  from urllib.parse import urlparse
+
+flags.DEFINE_string('maven_version', '3.6.3', 'The version of maven')
+FLAGS = flags.FLAGS
+MVN_URL = 'https://archive.apache.org/dist/maven/maven-{0}/{1}/binaries/apache-maven-{1}-bin.tar.gz'
+MVN_DIR = posixpath.join(INSTALL_DIR, "maven")
+MVN_ENV_PATH = "/etc/profile.d/maven.sh"
+
+MVN_ENV = """
+export JAVA_HOME={java_home}
+export M2_HOME={maven_home}
+export MAVEN_HOME={maven_home}
+export PATH={maven_home}/bin:$PATH
+"""
+
+
+def GetRunCommand(arguments):
+  """ Return Maven run command including proxy settings """
+  command = "source {} && mvn {}".format(MVN_ENV_PATH, arguments)
+  proxy_settings = ""
+  http_proxy_params = " -Dhttp.proxyHost={host} -Dhttp.proxyPort={port} "
+  https_proxy_params = " -Dhttps.proxyHost={host} -Dhttps.proxyPort={port} "
+
+  if FLAGS["http_proxy"].present:
+    parsed_url = urlparse.urlparse(FLAGS.http_proxy)
+    proxy_settings += http_proxy_params.format(host=parsed_url.hostname, port=parsed_url.port)
+
+  if FLAGS["https_proxy"].present:
+    parsed_url = urlparse.urlparse(FLAGS.https_proxy)
+    proxy_settings += https_proxy_params.format(host=parsed_url.hostname, port=parsed_url.port)
+
+  if proxy_settings:
+    command += proxy_settings
+  return command
+
+
+def AptInstall(vm):
+  _Install(vm)
+
+
+def YumInstall(vm):
+  vm.InstallPackages('which')
+  _Install(vm)
+
+
+def _Install(vm):
+  vm.Install('openjdk')
+
+  # Download and extract maven
+  maven_full_ver = FLAGS.maven_version
+  maven_major_ver = maven_full_ver[:maven_full_ver.index('.')]
+  maven_url = MVN_URL.format(maven_major_ver, maven_full_ver)
+  vm.RemoteCommand(('mkdir {0} && curl -L {1} | '
+                    'tar -C {0} --strip-components=1 -xzf -').format(MVN_DIR, maven_url))
+
+  # Get JAVA_HOME
+  out, _ = vm.RemoteCommand('readlink -f `which java`')
+  out = out.strip()
+  java_home = out[:out.index('/jre')]
+
+  # Set env variables for maven
+  maven_env = MVN_ENV.format(java_home=java_home, maven_home=MVN_DIR)
+  cmd = 'echo "{0}" | sudo tee -a {1}'.format(maven_env, MVN_ENV_PATH)
+  vm.RemoteCommand(cmd)
+  vm.RemoteCommand("source {0}".format(MVN_ENV_PATH))
```

Comment:
Should any of `MVN_ENV` environment variables go in `~/.bashrc`?

Reply:
@bvliu I have kept the `MVN_ENV` variables separate from `~/.bashrc` because they are only needed and sourced in the `GetRunCommand` function, which should be the only way to call maven, in order to automatically include the proxy vars.

Please let me know if you also want the `MVN_ENV` variables in `~/.bashrc` and I will include them there as well.